subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/configurationcache/ConfigurationCacheFixture.groovy

```diff
@@ -255,6 +255,12 @@ class ConfigurationCacheFixture {
             // Runs in quiet mode, and does not log anything
             return
         }
+
+        if (details.storeReason != null) {
```

Comment:
This is due to a new reason for cache invalidation added in the `ConfigurationCacheFingerprintChecker`. The two options for store action in a few lines below do not cover this new case.

I found some cases where the CC fixture was not used, possibly due to impossibility of providing a store-reason string. Those tests use a combination of `ConfigurationCacheBuildOperationsFixture` and output checking, instead of using `ConfigurationCacheFixture`. I don't see why we shouldn't use a single fixture for such tests at the very least, because the `ConfigurationCacheFixture` does more checks than the build-operation one.